runner/local/simple.go

```diff
@@ -137,12 +138,16 @@ func (r *simpleRunner) run(cmd *runner.Command, runId runner.RunId, doneCh chan
 	// Wait for checkout or cancel
 	select {
 	case <-doneCh:
+		go func() {
```

Comment:
I think I'm missing something :( As I'm reading it: if doneCh is closed, we start a goroutine that waits on checkoutDone and calls Release.  If doneCh is not closed, we just block until checkoutDone is closed and then defer the Release statement, so in both cases waiting for checkoutDone to be closed.